tests/kernel/device/src/main.c

```diff
@@ -221,152 +219,22 @@ void test_pre_kernel_detection(void)
 	}
 }
 
-#ifdef CONFIG_PM_DEVICE
 /**
- * @brief Test system device list query API with PM enabled.
+ * @brief Test system device list query API.
  *
  * It queries the list of devices in the system, used to suspend or
  * resume the devices in PM applications.
  *
  * @see z_device_get_all_static()
  */
-static void test_build_suspend_device_list(void)
+static void test_device_list(void)
 {
 	struct device const *devices;
 	size_t devcount = z_device_get_all_static(&devices);
 
 	zassert_false((devcount == 0), NULL);
 }
 
-/**
- * @brief Test APIs to enable and disable automatic runtime power management
- *
- * @details Test the API enable and disable, cause we do not implement our PM
- * API here, it will use the default function to handle power status. So when
- * we try to get power state by pm_device_state_get(), it will default
- * return power state zero. And we check it.
- *
- * @ingroup kernel_device_tests
- */
-static void test_enable_and_disable_automatic_runtime_pm(void)
-{
-	const struct device *dev;
-	int ret;
-	enum pm_device_state device_power_state;
-
-	dev = device_get_binding(DUMMY_PORT_2);
-	zassert_false((dev == NULL), NULL);
-
-	/* check its status at first */
-	/* for cases that cannot run runtime PM, we skip it now */
-	ret = pm_device_state_get(dev, &device_power_state);
-	if (ret == -ENOSYS) {
-		TC_PRINT("Power management not supported on device");
-		ztest_test_skip();
-		return;
-	}
-
-	zassert_true((ret == 0),
-		"Unable to get active state to device");
-
-	/* enable automatic runtime PM and check its status */
-	pm_device_runtime_enable(dev);
-	zassert_not_null((dev->pm), "No device pm");
-	zassert_true((dev->pm->enable), "Pm is not enable");
-
-	/* disable automatic runtime PM and check its status */
-	pm_device_runtime_disable(dev);
-	zassert_false((dev->pm->enable), "Pm shall not be enable");
```

Comment:
I don't seen an equivalent test that verifies `pm_device_runtime_enable/disable()` API.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/pm/device_runtime_api/src/main.c